process/block/metablock.go

```diff
@@ -555,14 +556,10 @@ func (mp *metaProcessor) indexBlock(
 
 	mp.indexer.UpdateTPS(mp.tpsBenchmark)
 
-	txPool := mp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock)
-	scPool := mp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock)
-
-	for hash, tx := range scPool {
-		txPool[hash] = tx
-	}
-	for hash, tx := range rewardsTxs {
-		txPool[hash] = tx
+	pool := &indexer.Pool{
+		Txs:     mp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock),
```

Comment:
this works as expected because the transactionCoordinator.GetAllCurrentUsedTxs create new maps on each call.

Reply:
Yes. This was used before and all the transactions were indexed.

Reply:
Yup

Reply:
no need to add type invalid and receipts?

Reply:
On the metachain we don't have receipts and invalid transactions. @sasurobert can you confirm this, please.